### PR TITLE
[refactor](Nereids)split rewrite and insert into memo to 2 functions

### PR DESCRIPTION
**Pull request by morrySnow**
## Problem Summary:

Split rewrite and insert into memo to 2 functions to make the code easy to read.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
@924060929 @englefly PTAL

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java [hunk 1]
```diff
@@ -122,44 +124,63 @@ private Plan groupToTreeNode(Group group) {
...
+            GroupExpression groupExpression, Group target, LogicalProperties logicalProperties) {
         GroupExpression existedGroupExpression = groupExpressions.get(groupExpression);
         if (existedGroupExpression != null) {
-            Group mergedGroup = existedGroupExpression.getOwnerGroup();
             if (target != null && !target.getGroupId().equals(existedGroupExpression.getOwnerGroup().getGroupId())) {
-                mergedGroup = mergeGroup(target, existedGroupExpression.getOwnerGroup());
-            }
-            if (rewrite) {
-                mergedGroup.setLogicalProperties(logicalProperties);
+                mergeGroup(target, existedGroupExpression.getOwnerGroup());
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java:142]: After mergeGroup(), the first argument `target`  may be removed from memo. 
`target` is input parameter, the caller may use it after `copyIn()`.
How about  `mergeGroup(existedGroupExpression.getOwnerGroup(), target);` ?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java [hunk 2]
```diff
@@ -122,44 +124,63 @@ private Plan groupToTreeNode(Group group) {
...
+            Group group = new Group(groupIdGenerator.getNextId(), groupExpression, logicalProperties);
+            groups.add(group);
+        }
+        groupExpressions.put(groupExpression, groupExpression);
+        return new Pair<>(true, groupExpression);
+    }
+
+    /**
+     * Rewrite groupExpression to target group.
+     * If group expression is already in memo and target group is not null, we replace logical properties.
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java:158]: this comment inconsistent with the code. the code is 'If group expression is already in memo, we replace logical properties regardless the target group present or not', and we can add the comment: 'for replace UnboundLogicalProperties to LogicalProperties'

**Review by englefly — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**